src/SAFE.Client/SAFE.fs

```diff
@@ -56,71 +58,96 @@ type ApiCall<'TStart, 'TFinished> =
 type RemoteData<'T> =
     /// The data has not yet started loading.
     | NotStarted
-    /// The data is now being loaded.
-    | Loading
+    /// The data is now being loaded. May already contain some data i.e. a refresh operation is in process.
+    | Loading of 'T option
     /// The data is available.
     | Loaded of 'T
 
     /// Unwraps the Loaded value, or returns the supplied default value.
     member this.DefaultValue v =
         match this with
         | NotStarted
-        | Loading -> v
+        | Loading None -> v
+        | Loading(Some value)
         | Loaded value -> value
 
-    /// Returns whether the `RemoteData<'T>` value has been loaded or not.
+    /// Returns whether the `RemoteData<'T>` value has been loaded.
     member this.HasLoaded =
         match this with
         | NotStarted
-        | Loading -> false
+        | Loading _ -> false
         | Loaded _ -> true
 
-    /// Returns whether the `RemoteData<'T>` value is loading or not.
+    /// Returns whether the `RemoteData<'T>` value has been loaded.
+    member this.HasData =
+        match this with
+        | NotStarted
+        | Loading None -> false
+        | Loading(Some _)
+        | Loaded _ -> true
+
+    /// Returns whether the `RemoteData<'T>` value is loading. This will return true for both first-time and refresh-style loads.
     member this.IsStillLoading =
         match this with
-        | Loading -> true
+        | Loading _ -> true
         | NotStarted
         | Loaded _ -> false
 
-    /// Returns whether the `RemoteData<'T>` value has started loading or not.
-    member this.HasStarted =
+    /// Returns whether the `RemoteData<'T>` value is refreshing itself i.e. Loading (Some _)
+    member this.IsRefreshing =
         match this with
-        | NotStarted -> false
-        | Loading
-        | Loaded _ -> true
+        | Loading(Some _) -> true
+        | _ -> false
+
+    /// Returns whether the `RemoteData<'T>` value has not yet started.
+    member this.HasNotStarted =
+        match this with
+        | NotStarted -> true
+        | Loading _
+        | Loaded _ -> false
 
     /// Maps the underlying value of the remote data, when it exists, into another shape.
     member this.Map mapper =
         match this with
         | NotStarted -> NotStarted
-        | Loading -> Loading
+        | Loading None -> Loading None
+        | Loading(Some value) -> Loading(Some(mapper value))
```

Comment:
I assume that you considered `Loading x -> Loading(Option.map mapper x))`? Either's fine, just wanted to flag the alternative as a possibility.

Reply:
Mmmm, that's more compact but not convinced about readability - let me try it in front of me and see how it looks